eng/native/build-commons.sh

```diff
@@ -104,13 +105,25 @@ build_native()
                 exit "$exit_code"
             fi
         else
-            # Generate the dummy version.c, but only if it didn't exist to make sure we don't trigger unnecessary rebuild
+            # Generate the dummy version.c and runtime_version.h, but only if they didn't exist to make sure we don't trigger unnecessary rebuild
             __versionSourceLine="static char sccsid[] __attribute__((used)) = \"@(#)No version information produced\";"
             if [[ -e "$__versionSourceFile" ]]; then
                 read existingVersionSourceLine < "$__versionSourceFile"
             fi
             if [[ "$__versionSourceLine" != "$existingVersionSourceLine" ]]; then
                 echo "$__versionSourceLine" > "$__versionSourceFile"
```

Comment:
Nit: I would move this to be after `cat << EOF > $runtimeVersionHeaderFile`. It will handle the situation where the build is aborted and restarted better.

Reply:
Sure, but I didn't quite follow how would reordering make a difference in this case? If condition is met, both file writes will be executed, otherwise neither.

Reply:
> both file writes will be executed, 

Unless somebody manages to Ctrl+C the build between the two writes.

Reply:
I have reordered the lines.

> Unless somebody manages to Ctrl+C the build between the two writes.

I think the complexity of `echo $text > $file` and `cat << EOF > $runtimeVersionHeaderFile $multilineText` is same as long text fits into a single pagesize so Ctrl+C effect in either order is equally likely.